build/wd_test.bzl

```diff
@@ -11,12 +14,35 @@ def wd_test(
         specified.) The extension `.wd-test` is also permitted instead of `.capnp`, in order to
         avoid confusing other build systems that may assume a `.capnp` file should be compiled. As
         an extension, `.gpu-wd-test` is supported to enable special handling for GPU tests.
-     data: Files which the .capnp config file may embed. Typically JavaScript files.
+     srcs: Additional source files that can be compiled if they are typescript to be consumed by the test.
+     data: Additional Files which the .capnp config file may embed.
      args: Additional arguments to pass to `workerd`. Typically used to pass `--experimental`.
     """
 
+    for d in data:
+        if d.endswith(".js"):
+            fail("javascript files should be passed as srcs")
+
     # Add workerd binary to "data" dependencies.
-    data = data + [src, "//src/workerd/server:workerd"]
+    data = data + [src, "//src/workerd/server:workerd"] + srcs
+
+    ts_srcs = [src for src in srcs if src.endswith(".ts")]
```

Comment:
Is there a reason you need the new `srcs` argument at all? Why not just filter `data` for arguments that end in `*.ts`? It seems simpler to me.

Reply:
[ ] Honestly it was cleaner to help me debug at first, I can collapse them.